pkg/controllers/cassandra/cluster_control.go

```diff
@@ -210,8 +215,21 @@ func NextAction(c *v1alpha1.CassandraCluster) controllers.Action {
 			return &actions.ScaleOut{
 				Cluster:  c,
 				NodePool: &np,
-			}
+			}, nil
+		}
+
+		statefulSetName := util.NodePoolResourceName(c, &np)
+		ss, err := statefulSetLister.StatefulSets(c.Namespace).Get(statefulSetName)
+		if err != nil {
+			return nil, err
+		}
+
+		if *np.Replicas < *ss.Spec.Replicas {
+			return &actions.ScaleIn{
+				Cluster:  c,
+				NodePool: &np,
+			}, nil
```

Comment:
I'm keen to keep this function unchanged :-)
I think it should be possible to instead check `nps.ReadyReplicas` here...if not, let's discuss.